ADR/0009-pipeline-service-via-operator.md

```diff
@@ -0,0 +1,74 @@
+# 0009 Pipeline Service via Operator
+
+Created: 2023-09-29
+Last Updated: 2023-09-29
+
+## Status
+
+Proposed
+
+## Context
+
+kcp is no longer being used as a control plane for RHTAP. This means that
+"Pipeline Service" cannot be deployed as an independent service. For our
+initial MVP, all Tekton APIs need to be deployed onto a standard OpenShift
+cluster (specifically OpenShift Dedicated).
+
+## Decision
+
+All Tekton APIs will be provided using the stock OpenShift Pipelines operator (OSP).
+In the spirit of developing in a "Service First" manner, RHTAP will deploy
+a candidate "nightly" release of the operator. The service will be defined in
+the [pipeline-service](https://github.com/openshift/pipeline-service)
+repository, which is then imported into
+[infra-deployments](https://github.com/redhat-appstudio/infra-deployments) as
+an ArgoCD application.
+
+Not all metrics required for operating the service are exposed natively by the
+controllers. The `pipeline-metrics-exporter` controller is to be used as a test
+bed to expose new metrics, with the goal of upstreaming those metrics as they
+mature and prove their value.
+
+Configurations that are specific to RHTAP must be made available through
+the OSP CRDs. The following changes are specific to RHTAP:
+
+- Disable Tekton Triggers. Pipelines as Code will be the favored mechanism for
+  event-based triggering of pipelines for now. This decision can be revisited
+  in the future based on need.
+- Disable the pruner that comes with the operator. Tekton Results will be used
+  to prune `PipelineRun` and `TaskRun` data off cluster, thereby ensuring data
+  is archived before removal.
+- Direct Pipelines as Code to use a URL pattern that displays the `PipelineRun`
+  or `TaskRun` info from the Hybrid Application Console (HAC). This ensures
+  end users do not need access to the underlying compute cluster(s).
+- The Pipelines as Code application name must match the GitHub Application name, so that users understand which GitHubApplication is responsible for triggering the pipelines.
+- The GitHub Application secret value, deployed using an ExternalSecret.
+- Any configuration related to performance.
+
+Furthermore, as the service will be accessed through CodeReadyToolchain (CRT), the
+following changes are also specific to RHTAP:
+- Deploying a proxy (known as `SprayProxy`) on the CRT host cluster that redirects
```

Comment:
(nit) maybe not part of this PR but it would be nice there was a document on how the redirection mechanism works, what is the reasoning behind it and if there is further plans to improve this..

Reply:
I've opened a story so we create a ADR (or update this one) to include more information about the SprayProxy.